go-app-ussd_popi_user_data.js

```diff
@@ -319,16 +325,35 @@ go.app = function() {
                 ],
                 next: function(choice) {
                     return self.im.user
-                    // TODO: update on IdentityStore
                     .set_lang(choice.value)
                     .then(function() {
-                        self.im.user.set_answer("state_language", choice.value);
-                        return 'state_updated';
+                        self.im.user.set_answer("new_language", choice.value);
+                        return 'state_switch_lang';
                     });
                 },
             });
         });
 
+        self.add('state_switch_lang', function(name) {
+            var change_info = {
+                "registrant_id": self.im.user.answers.operator.id,
+                "action": "momconnect_change_language",
+                "data": {
+                    "language": self.im.user.answers.new_language,
```

Comment:
ah ok, I though you had to explicitly set the answers for them to persist through states.